src/types.rs

```diff
@@ -0,0 +1,66 @@
+use derive_more::derive::Display;
+use serde::Serialize;
+use sqlx::Type;
+
+#[derive(Type, Debug, PartialEq, Eq, Serialize)]
+#[sqlx(type_name = "chain_status_type", rename_all = "lowercase")]
+pub enum ChainStatus {
+  Canonical,
+  Pending,
+  Orphaned,
+}
+
+#[derive(Type, Debug, PartialEq, Eq, Serialize)]
+#[sqlx(type_name = "command_type", rename_all = "lowercase")]
+pub enum UserCommandType {
+  Payment,
+  Delegation,
+}
+
+#[derive(Type, Debug, PartialEq, Eq, Serialize)]
+#[sqlx(type_name = "internal_command_type", rename_all = "snake_case")]
+pub enum InternalCommandType {
+  FeeTransferViaCoinbase,
+  FeeTransfer,
+  Coinbase,
+}
+
+#[derive(Type, Debug, PartialEq, Eq, Serialize, Display, Clone)]
+#[sqlx(type_name = "transaction_status", rename_all = "lowercase")]
+pub enum TransactionStatus {
+  Applied,
+  Failed,
+}
+
+#[derive(Debug, Display)]
+pub enum OperationStatus {
+  Success,
+  Failed,
+}
+
+impl From<TransactionStatus> for OperationStatus {
+  fn from(status: TransactionStatus) -> Self {
+    match status {
+      TransactionStatus::Applied => OperationStatus::Success,
+      TransactionStatus::Failed => OperationStatus::Failed,
+    }
+  }
+}
+
+#[derive(Debug, Display)]
+pub enum OperationType {
+  FeePayerDec,
+  FeeReceiverInc,
+  CoinbaseInc,
+  AccountCreationFeeViaPayment,
+  AccountCreationFeeViaFeePayer,
+  AccountCreationFeeViaFeeReceiver,
+  PaymentSourceDec,
+  PaymentReceiverInc,
+  FeePayment,
+  DelegateChange,
+  CreateToken,
+  MintTokens,
+  ZkappFeePayerDec,
+  ZkappBalanceUpdate,
```

Comment:
I suppose some operation types are now deprecated (see: https://github.com/MinaProtocol/mina/pull/15907/files#diff-74481279bb056609148ee774a64cb4a6fae9ee0cd9a74345cef99eb3f760909b) Can you confirm @joaosreis?